system/lib/wasmfs/syscalls.cpp

```diff
@@ -386,6 +405,11 @@ long __syscall_mkdir(long path, long mode) {
   if (curr) {
     return -EEXIST;
   } else {
+    // Mask rwx permissions for user, group and others, and the sticky bit.
+    // This prevents users from entering S_IFREG for example.
+    // https://www.gnu.org/software/libc/manual/html_node/Permission-Bits.html
+    // This is a placeholder for now which matches the JS filesystem.
```

Comment:
We determined that this was important to prevent users from setting bits that they shouldn't have control over.
```suggestion
```